subprojects/core/src/main/java/org/gradle/api/internal/artifacts/configurations/ConfigurationRoles.java

```diff
@@ -44,6 +44,11 @@ private ConfigurationRoles() {
     @Deprecated
     public static final ConfigurationRole ALL = createNonDeprecatedRole("Legacy", true, true, true);
 
+    /**
+     * A configuration that is retired and not meant to be used for any purpose.  This is primarily useful in creating a migrating configuration role in {@link ConfigurationRolesForMigration}.
+     */
+    public static final ConfigurationRole NONE = createNonDeprecatedRole("Deprecated", false, false, false);
```

Comment:
🤔 It's a little weird to see `createNONDEPRECATEDRole` with a comment saying it's effectively deprecated.  Can this live in `ConfigurationRolesForMigration` instead?

Reply:
Yeah, that makes sense - I've moved this to `ConfigurationRolesForMigration`, which makes sense since you'd only really want this during migration.